### ISSUE TITLE
`zig c++`: Compile CUDA code

### ISSUE THREAD

**Issue by saierd** [ziglang/zig#10634]
### Zig Version

0.10.0-dev.290+3901b6fb0

### Steps to Reproduce

Try to compile a `.cu` file with `zig c++`.

### Expected Behavior

`clang++` can compile CUDA code as described here: https://llvm.org/docs/CompileCudaWithLLVM.html It would be nice if zig could also expose this functionality.

I am not sure if this is a simple matter of adding the file extension to treat it as compilable code or if there needs to be some other logic for forwarding all of the additional arguments.

### Actual Behavior

zig tries to link using the given arguments, because it does not know the file extension.

**Comment by andrewrk** [comment]
Looks like your PR (landed in 2f41bd3be438dae2a188cfae3295dc28f4e9d434) solved the issue. Let me know if you think there is still an issue.

**Comment by saierd** [comment]
No, it did not really solve this issue. As I explained in [this comment](https://github.com/ziglang/zig/pull/10704#issuecomment-1023616464) the PR allows to use `zig cc` as a host compiler for `nvcc`. `clang`'s own CUDA mode does not work yet.

I just tried to run `zig c++ main.cu --cuda-gpu-arch=sm_35 -mcpu=sm_35` on a small example as described in the link above, but it failed with

```
error: Unknown CPU: 'sm_35'
```
